docs/core-features/non_linear_operations.md

```diff
@@ -0,0 +1,158 @@
+# Non-linear operations
```

Comment:
I'm not sure of your intention here, but from what I understand the goal is to push details or hard things to understand more deeper in the doc and have a quick and first overview of features. If that the purpose of this documentation refactoring I think we should even not talk about TFHE and lookup table here, just listing operations and at the end of each one give a link to the detailed documentation with different strategies etc...

Reply:
Maybe just saying there are two classes of operations linear one that is fast and the other one, then later explain why..?

Reply:
@BourgerieQuentin , do you prefer the new wording?